lib/service.js

```diff
@@ -3091,7 +3104,99 @@
+
+                // 2. Filter the files
+                var files = response 
+                                && response.body 
+                                && response.body.entry 
+                                && response.body.entry.filter(f => f.name === filename);
+                
+                // 3. Check if the file exists
+                if(files && files.length == 0) {
+                    return { code:"FILE_NOT_FOUND", message: `File with the name '${filename}' does not exist.`};
+                }
+
+                // 4. Create a local instance
+                var configurationFile = new root.ConfigurationFile(that.service, filename);
+
+                // 5. Load the file content
+                var fetchedFile = await configurationFile.fetchAsync();
+
+                return {file: fetchedFile};
+            }
+            catch(error) {
+                return { errorCode:"SYSTEM_ERROR", errorMessage: error};
+            }
+        },
+
+        doesStanzaExist: async function(file, stanza) {
+            try {
+                
+                // 1. check if the stanza exists
+                var fetchedStanza = file.item(stanza);
+                
+                if(!fetchedStanza) {
+                    return { code:"STANZA_NOT_FOUND", message: `Stanza with the name '${stanza}' does not exist.`};
+                }
+                else {
+                    return {stanza: fetchedStanza};
+                }
+            }
+            catch(error) {
+                return { errorCode:"SYSTEM_ERROR", errorMessage: error};
+            }
+        },
+
+        createAsync: async function (filename, stanza, keyValueMap, callback) {
+            callback = callback || function() {};
+            var that = this;
+
+            // 1. Check if the file exists
+            var doesFileExistResponse = await this.doesFileExist(filename);
+            var configFile = doesFileExistResponse.file;
+
+            // 2. If the file doesn't exist, create a new file
+            if(!configFile) {
+                console.log(doesFileExistResponse);
+                console.log("Creating a new configuration file...");
+
+                that.create( { __conf: filename }, 
+                    (err, newFile) => console.log(`Configuration File created: ${filename}`) );
+
+                configFile = new root.ConfigurationFile( that.service, filename );
+                configFile = await configFile.fetchAsync();
+            }
+
+            // 2. Check if the stanza exists
+            var doesStanzaExistResponse = await this.doesStanzaExist(configFile, stanza);
+            var configStanza = doesStanzaExistResponse.stanza;
+
+            // 3. If the stanza doesn't exist, create a new stanza with given keyValueMap
+            if(!configStanza) {
+                console.log(doesStanzaExistResponse);
+                console.log("Creating a new configuration stanza...");
```

Comment:
We'll want to remove the console.log before merging

Reply:
Do we need all console.log to be removed or just the once like line#3184?

Reply:
I think all of them that have been added in this PR - did you notice others within the SDK?

Reply:
Done